### PR TITLE
Move common http test utilities from testutils to common4j test fixtures

### PR DESCRIPTION
**Pull request by shahzaibj**
Description: Move common http test utilities from testutils to common4j test fixtures.

Why we are moving these classes? We are moving these classes to test fixtures because test fixtures is where they truly belong. The primary purpose of test fixtures is to be able to share test code across modules. 

This is to break https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1770 into multiple smaller PRs.

### CODE REVIEW

**Review by rpdome — APPROVED** [review]

**Comment by iamgusain** [comment]
Would be nice to update the description with what is being changed here

**Comment by shahzaibj** [comment]
> Would be nice to update the description with what is being changed here

I've added the title to description as well. That's basically all there is to this PR. I'm not sure what more I'd put for a simple PR like this that's just relocated some files.

File: common4j/build.gradle [hunk 1]
```diff
@@ -187,10 +213,15 @@ dependencies {
...
+
+    testFixturesCompileOnly 'com.github.spotbugs:spotbugs-annotations:3.1.12'
+    testFixturesCompileOnly "org.projectlombok:lombok:$rootProject.ext.lombokVersion"
+    testFixturesAnnotationProcessor "org.projectlombok:lombok:$rootProject.ext.lombokVersion"
+    resolvableTestFixturesImplementation 'org.robolectric:junit:4.5.1'
 }
 
-sourceCompatibility = "1.7"
-targetCompatibility = "1.7"
+sourceCompatibility = "1.8"
```
Comment (iamgusain) [inline common4j/build.gradle:223]: Why is this change needed?

**Review by iamgusain — APPROVED** [review]

**Comment by iamgusain** [comment]
> > Would be nice to update the description with what is being changed here
> 
> I've added the title to description as well. That's basically all there is to this PR. I'm not sure what more I'd put for a simple PR like this that's just relocated some files.

Something like why we are moving the classes will add some context, when we look back at the PR later.

**Comment by shahzaibj** [comment]
> > > Would be nice to update the description with what is being changed here
> > 
> > 
> > I've added the title to description as well. That's basically all there is to this PR. I'm not sure what more I'd put for a simple PR like this that's just relocated some files.
> 
> Something like why we are moving the classes will add some context, when we look back at the PR later.

Added this:

Why we are moving these classes? We are moving these classes to test fixtures because test fixtures is where they truly belong. The primary purpose of test fixtures is to be able to share test code across modules.

Test Fixtures is a concept that @p3dr0rv had introduced to the team some time ago, and I think I also covered it again in one of my recent brown-bags as well.

**Review by tanmaymanolkar1 — APPROVED** [review]

File: common4j/build.gradle (hunk 1 above)
Comment (shahzaibj) [reply common4j/build.gradle:223]: I think I needed it for some other code I wrote in the bigger PR. So this is a remnant from that but since it belonged in `build.gradle` therefore I included it here as all the `build.gradle` changes were here so it was easier for me to organize the PRs. 

Also I think we've actually already technically been on Java 8 since Melissa's YubyKit changes (YubyKit SDK is on Java 8). We just hadn't explicitly made the change in build.gradle here.
